flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/CurrentJobsOverviewHandler.java

```diff
@@ -92,24 +101,26 @@ public CurrentJobsOverviewHandler(
 					StringWriter writer = new StringWriter();
 					try {
 						JsonGenerator gen = JsonFactory.JACKSON_FACTORY.createGenerator(writer);
+						final JobDetails.JobDetailsSerializer jobDetailsSerializer = new JobDetails.JobDetailsSerializer();
+
 						gen.writeStartObject();
 
 						if (includeRunningJobs && includeFinishedJobs) {
 							gen.writeArrayFieldStart("running");
-							for (JobDetails detail : result.getRunningJobs()) {
-								writeJobDetailOverviewAsJson(detail, gen, now);
+							for (JobDetails detail : result.getRunning()) {
+								jobDetailsSerializer.serialize(detail, gen, null);
 							}
 							gen.writeEndArray();
 
 							gen.writeArrayFieldStart("finished");
-							for (JobDetails detail : result.getFinishedJobs()) {
-								writeJobDetailOverviewAsJson(detail, gen, now);
+							for (JobDetails detail : result.getFinished()) {
+								jobDetailsSerializer.serialize(detail, gen, null);
 							}
 							gen.writeEndArray();
 						} else {
```

Comment:
On a second look, I think this response variant is not covered by the ported handler.

Reply:
Yes, because I think it's not so nice to return different answer formats depending on how the handler was instantiated. The respective information can be easily extracted from the complete overview. Moreover, I think we never used these endpoints in the first place.